starlette/middleware/base.py

```diff
@@ -5,6 +5,12 @@
 from starlette.responses import Response, StreamingResponse
 from starlette.types import ASGIApp, Receive, Scope, Send
 
+try:
+    from asyncio.exceptions import InvalidStateError  # type: ignore
+except ImportError:  # pragma: nocover
+    from asyncio import InvalidStateError
```

Comment:
From the docs it seems like `asyncio.InvalidStateError` should be available on 3.6-3.8? Which versions isn't it available on?

Reply:
Ah, good point. This was a reactionary change based on the failure in 3.6 and 3.7. I originally had it as `asyncio.exceptions.InvalidStateError`, which isn't valid in the earlier versions. That's a good point, though. I can import it from the top-level for all versions.

Reply:
Actually, with your `if task.done()` the import can be removed entirely.